### PR TITLE
[PREVIEW] Add JA3S fingerprint support

### PR DESCRIPTION
**Pull request by thus**
This is just a preview, so please don't merge it!

This pull request implements JA3S fingerprints as implemented in the bro script in the JA3 github repo:
https://github.com/salesforce/ja3/blob/master/bro/ja3s.bro

Redmine:
https://redmine.openinfosecfoundation.org/issues/2684

Prscript:
- PR thus-pcap: https://buildbot.openinfosecfoundation.org/builders/thus-pcap/builds/178
- PR thus: https://buildbot.openinfosecfoundation.org/builders/thus/builds/178

### CODE REVIEW

**Comment by thus** [comment]
Thanks, Victor! I'll change the code and resubmit :)

I talked with one of the JA3 authors. It seems the 's' in JA3s is in lowercase, so I will have to change that a few places too!

**Comment by victorjulien** [comment]
Great, looking forward to the next iteration!

File: src/detect-tls-ja3s-hash.c [hunk 1]
```diff
@@ -0,0 +1,378 @@
+/* Copyright (C) 2017 Open Information Security Foundation
```
[the remark this answers is not in the document]
Comment (thus) [reply src/detect-tls-ja3s-hash.c]: 2019 now ;)

File: src/detect-tls-ja3s-hash.c [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+            SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+        }
+
+        if (cd->content_len == 32)
+            return TRUE;
+
+        *sigerror = "Invalid length of the specified JA3S hash (should "
+                    "be 32 characters long). This rule will therefore "
+                    "never match.";
+        SCLogWarning(SC_WARN_POOR_RULE,  "rule %u: %s", s->id, *sigerror);
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-tls-ja3s-hash.c]: I think this should be turned into an error.

File: src/detect-tls-ja3s-hash.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-tls-ja3s-hash.c]: :)

**Comment by thus** [comment]
Replaced by #3850

**Pull request closed without merging**